lib/http/static.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+const {
+  ArrayPrototypeIncludes,
+  Promise,
+  RegExpPrototypeExec,
+  StringPrototypeEndsWith,
+  StringPrototypeIncludes,
+  StringPrototypeIndexOf,
+  StringPrototypeLastIndexOf,
+  StringPrototypeReplace,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+  StringPrototypeToLocaleLowerCase,
+} = primordials;
+
+const { cwd, nextTick } = process;
+const console = require('internal/console/global');
+const { createReadStream } = require('fs');
+const { Server } = require('_http_server');
+const { URL, isURLInstance, pathToFileURL } = require('internal/url');
+const { emitExperimentalWarning, kEmptyObject } = require('internal/util');
+const { getOptionValue } = require('internal/options');
+const { validateFunction, validatePort } = require('internal/validators');
+
+if (module.isPreloading) {
+  const requiredModules = getOptionValue('--require');
+  if (ArrayPrototypeIncludes(requiredModules, `node:${module.id}`) ||
+      ArrayPrototypeIncludes(requiredModules, module.id)) {
+    const { parseArgs } = require('internal/util/parse_args/parse_args');
+    const { values: { port, host } } = parseArgs({ options: {
+      port: {
+        type: 'string',
+        short: 'p',
+      },
+      host: {
+        type: 'string',
+        short: 'h',
+      },
+    } });
+    nextTick(createStaticServer, { directory: process.argv[1], port, host });
+    process.env.NODE_REPL_EXTERNAL_MODULE = 'node:http/static';
+    process.argv[1] = 'node:http/static';
+  }
+}
+
+function filterDotFiles(url, fileURL) {
+  return !StringPrototypeIncludes(fileURL.pathname, '/.');
+}
+
+function createStaticServer(options = kEmptyObject) {
+  emitExperimentalWarning('http/static');
+
+  const {
+    directory = cwd(),
```

Comment:
To be safe we likely should also forbid the root directory containing a dot, to help prevent the possibility, for instance, of some malicious party starting a static server on paths like ~/.ssh etc

Reply:
Added test in https://github.com/nodejs/node/pull/45096/commits/fb2d3498f4ee48ce7d77e4d7d85ac65d2eb9ec80.